app/api/available-models/route.ts

```diff
@@ -0,0 +1,83 @@
+import OpenAI from "openai";
+import { Ollama } from "ollama";
+
+export const dynamic = "force-dynamic";
+
+export async function POST(req: Request) {
+  try {
+    const { openaiApiKey, ollamaEndpoint } = await req.json();
+
+    if (!openaiApiKey && !ollamaEndpoint) {
+      return new Response(
+        JSON.stringify({ models: { openai: [], ollama: [] } }),
+        {
+          status: 200,
+          headers: { "Content-Type": "application/json" },
+        },
+      );
+    }
+
+    let availableOpenaiModels,
+      availableOllamaModels: string[] = [];
+
+    if (openaiApiKey) {
+      const openai = new OpenAI({ apiKey: openaiApiKey });
+      const openaiModels = openai.models.list();
+      const openaiModelsResponse = await openaiModels;
+
+      if (openaiModelsResponse.data) {
+        availableOpenaiModels = openaiModelsResponse.data
+          .filter(
+            (model) =>
+              (model.id.includes("gpt-4o") ||
+                model.id.includes("gpt-4-turbo")) &&
+              !model.id.includes("preview"),
+          )
+          .map((model) => model.id);
+      }
+
+      if (ollamaEndpoint) {
```

Comment:
This ollamaEndpoint check happens only when there is an openAIKey. I think this should probably be moved out of the openaikey check.

Reply:
Ahh good catch. Fixed it.